subsys/net/lib/config/init.c

```diff
@@ -379,6 +380,17 @@ int net_config_init_by_iface(struct net_if *iface, const char *app_info,
 #endif
 	}
 
+#if defined(CONFIG_NET_VLAN) && (CONFIG_NET_CONFIG_MY_VLAN_ID > 0)
+	int ret = net_eth_vlan_enable(iface,
+				      CONFIG_NET_CONFIG_MY_VLAN_ID);
+
+	if (ret < 0) {
+		NET_ERR("Network interface %d (%p): cannot set VLAN tag (%d)\n",
+			net_if_get_by_iface(iface), iface, ret);
+		return -ENOEXEC;
+	}
+#endif
```

Comment:
Should we return if the VLAN fails? 
Also maybe move it outside of this function like it is done with `CONFIG_NET_CONFIG_MY_IPV4_ADDR`:

```c
#if defined(CONFIG_NET_VLAN) && (CONFIG_NET_CONFIG_MY_VLAN_ID > 0)
static void setup_vlan(struct net_if *iface)
{
	int ret = net_eth_vlan_enable(iface, CONFIG_NET_CONFIG_MY_VLAN_ID);

	if (ret < 0) {
		NET_ERR("Network interface %d (%p): cannot set VLAN tag (%d)\n",
			net_if_get_by_iface(iface), iface, ret);
	}
}
#else
#define setup_vlan(...)
#endif /* CONFIG_NET_VLAN && (CONFIG_NET_CONFIG_MY_VLAN_ID > 0) */
```

And simply call `setup_vlan(iface)` in `net_config_init_by_iface`

Reply:
I could do it, but does it make sense do start all the ip/dhcp services if you fail to set the VLAN? they would fail because you'll end up in a network where your packets are not going anywere

Reply:
Clearly if you strongly object to that, I will do the changes, just let me know

Reply:
I think @rlubos could decide if this should abort the configuration.

Reply:
Well, it depends on whether VLAN configuration failure is recoverable - if user decides to retry VLAN configuration later with shell for instance, he could expect to have IPv4 already configured.

Another case, when there are more interfaces in the system, `net_if_get_default()` may not be the default one, so it'd still make sense to proceed with the configuration if VLAN setup fails (it verifies if it deals with Ethernet iface first). So IMO, we should be more permissive here.

And +1 for adding helper function.

Reply:
Ok I will do the modifications